### PR TITLE
chore: Adds screenshot artifact to iOS e2e

### PR DESCRIPTION
**Pull request by Ubax**
## Description

When iOS e2e test fail save screenshots as artifacts

## Checklist
- [ ] Ensured that CI passes

### CODE REVIEW

File: .github/workflows/ios-e2e-test.yml [hunk 1]
```diff
@@ -42,3 +42,8 @@ jobs:
       - name: Test app
         working-directory: ${{ env.WORKING_DIRECTORY }}
         run: yarn test-e2e-ios
+      - uses: actions/upload-artifact@v2
+        if: ${{ failure() }}
+        with:
+          name: android-fail-screen-shoots
```
Comment (WoLewicki) [inline .github/workflows/ios-e2e-test.yml:48]: Why is there `android` if the PR mentions `iOS`?

Comment (Ubax) [reply .github/workflows/ios-e2e-test.yml:48]: A copy paste typo 🤦

File: .github/workflows/ios-e2e-test.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: Test iOS e2e
 on:
   pull_request:
     paths:
-        - 'Example/**'
+      - 'Example/**'
```
Comment (kacperkapusciak) [inline .github/workflows/ios-e2e-test.yml:5]: A question that should've been to previous PRs but I couldn't do a review in time before merging - does this mean e2e tests are only run on PRs when the `Example/` project changes? Shouldn't it also be run when `ios/` and `android/` are being changed? cc @WoLewicki

Comment (WoLewicki) [reply .github/workflows/ios-e2e-test.yml:5]: Yeah, for sure these tests should be run for such changes.

Comment (Ubax) [reply .github/workflows/ios-e2e-test.yml:5]: It takes about 40 minutes for tests to run. I am not sure if we want them ran on every change in most PRs. They will always be run on the master after merging. Also one can ran them locally if they are not sure if something works

Comment (WoLewicki) [reply .github/workflows/ios-e2e-test.yml:5]: Only master option seems enough for me.

Comment (kacperkapusciak) [reply .github/workflows/ios-e2e-test.yml:5]: Okay, running e2e on every commit is overkill but if it's possible we could maybe run these tests 1 time a day (in the night) on a hanging PR? If it's not possible it's okay (not too sure if GitHub Actions supports these kind of tricks). Our master is not directly production so it should be fine as you've said.

Comment (Ubax) [reply .github/workflows/ios-e2e-test.yml:5]: Unfortunately we can always do cron actions on master branch (or other specified in the workflow).

**Pull request merged**